gk/fib.c

```diff
@@ -1572,40 +1572,32 @@ check_prefix_security_hole_locked(struct ip_prefix *prefix,
 	return 0;
 }
 
+/*
+ * Returns:
+ *   1 if the prefix already exists
+ *   0 if the prefix does not exist
+ *  <0 if an error occurred
+ */
 static int
-check_prefix_locked(struct ip_prefix *prefix, enum gk_fib_action action,
-	struct gk_config *gk_conf)
+check_prefix_exists_locked(struct ip_prefix *prefix, struct gk_config *gk_conf)
 {
 	struct gk_lpm *ltbl = &gk_conf->lpm_tbl;
-	int ip_prefix_present;
 	uint32_t fib_id;
 
-	/* Make sure an entry doesn't already exist for this prefix. */
 	if (prefix->addr.proto == RTE_ETHER_TYPE_IPV4) {
-		ip_prefix_present = rte_lpm_is_rule_present(ltbl->lpm,
+		return rte_lpm_is_rule_present(ltbl->lpm,
 			ntohl(prefix->addr.ip.v4.s_addr),
 			prefix->len, &fib_id);
-		if (ip_prefix_present != 0) {
-			GK_LOG(ERR, "Prefix already exists in the FIB table (or can't be checked)\n");
-			return -1;
-		}
 	} else if (likely(prefix->addr.proto == RTE_ETHER_TYPE_IPV6)) {
-		ip_prefix_present = rte_lpm6_is_rule_present(ltbl->lpm6,
+		return rte_lpm6_is_rule_present(ltbl->lpm6,
 			prefix->addr.ip.v6.s6_addr,
 			prefix->len, &fib_id);
-		if (ip_prefix_present != 0) {
-			GK_LOG(ERR, "Prefix already exists in the FIB table (or can't be checked)\n");
-			return -1;
-		}
 	} else {
```

Comment:
Style: we no longer need the `else if` or the `else` clause. The `else if` clause can become an independent `if` statement, and the `else` clause no longer needs an enclosure.

Reply:
In **fib: allow Grantor entries to be updated**, the structure of this function reverts back to not having `return` statements in each conditional, so I'll leave it the way it is in this patch.